### PR TITLE
Add support for games that operate on hashes

### PR DESCRIPTION
**Pull request by vadi2**
Add support for games that operate on hashes. The basic premise is to pre-create rooms that don't exist yet to assign them a hash, but still consider them "non-existent" for all intents and purposes to keep the existing logic happy.

### CODE REVIEW

File: mudlet-mapper.xml [hunk 1]
```diff
@@ -11821,6 +11917,104 @@ end</script>
...
+    Temple                      = 57,
+    Tower                       = 79,
+    transportation              = 48,
+    Tundra                      = 16,
+    Underworld                  = 28,
+    Urban                       = 8,
+    Valley                      = 13,
+    Villa                       = 75,
+    Village                     = 47,
+    Village                     = 55,
```
Comment (keneanung) [inline mudlet-mapper.xml:12001]: There's a duplication here

**Comment by araskal** [comment]
any chance this can be pulled into the master branch?

**Comment by vadi2** [comment]
@araskal have a look at testing this out to see if it still works. Note that this doesn't go the full way to add support for the game - so it's probably missing many triggers and etc, only support for using hashes as room identifiers.

File: mudlet-mapper.xml [hunk 2]
```diff
@@ -4546,7 +4545,7 @@ mmp.settings:setOption(matches[3], val)</script>
 				<packageName></packageName>
 				<regex>^area list$</regex>
 			</Alias>
-			<AliasGroup isActive="no" isFolder="yes">
+			<AliasGroup isActive="yes" isFolder="yes">
```
Comment (keneanung) [inline mudlet-mapper.xml:4548]: We probably don't want mapping enabled by default.
```suggestion
			<AliasGroup isActive="no" isFolder="yes">
```

File: mudlet-mapper.xml [hunk 3]
```diff
@@ -10359,7 +10375,14 @@ end
 					<packageName></packageName>
 					<script>function mmp.centerRoominfo()
   -- lusternia has gmcp.Room.Players before gmcp.Room.Info is created
-  if gmcp.Room.Info then centerview(gmcp.Room.Info.num) end
+  if gmcp.Room.Info then
+    if gmcp.Room.Info.num then
+      centerview(gmcp.Room.Info.num)
+    elseif gmcp.Room.Info.roomhash then
+    print(tostring(mmp.roomidFromHash(gmcp.Room.Info.roomhash)))
```
Comment (keneanung) [inline mudlet-mapper.xml:10382]: This looks like a leftover debug echo:
```suggestion
```

**Review by keneanung — APPROVED** [review]
Looks good. There are two things I noticed, with one which may cause confusion

**Pull request merged**